llvm/tools/llvm-profgen/ProfiledBinary.h

```diff
@@ -428,6 +431,14 @@ class ProfiledBinary {
 
   bool usePseudoProbes() const { return UsePseudoProbes; }
   bool useFSDiscriminator() const { return UseFSDiscriminator; }
+  bool isKernel() const { return IsKernel; }
+
+  bool isKernelImageName(StringRef BinaryName) const {
```

Comment:
Why do we need a `BinaryName` parameter given `ProfiledBinary` has a name associated with it (`getName`)?

Reply:
getName() would return the binary name of the kernel image, typically vmlinux, or a name with version info, like vmlinux-6.8.0-smp-1201.1.889.0. The image name from perf command is in of the forms in this function.

Reply:
Ok, then this is static function?

Reply:
Right. I'll make it static.